### PR TITLE
Use discretize.utils.active_from_xyz to get active topography cells

### PR DESCRIPTION
**Pull request by dccowan**

### CODE REVIEW

File: SimPEG/utils/model_utils.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
-from scipy.interpolate import griddata, interp1d
-from scipy.interpolate import NearestNDInterpolator, LinearNDInterpolator, interp1d
+from scipy.interpolate import griddata
 from scipy.spatial import cKDTree
 import scipy.sparse as sp
+from discretize.utils import active_from_xyz
+import warnings
 
 
 def surface2ind_topo(mesh, topo, gridLoc="CC", method="nearest", fill_value=np.nan):
```
Comment (jcapriot) [inline SimPEG/utils/model_utils.py:10]: Make surface2ind_topo return the same thing it did before please. Even though it is deprecated, it's functionality should be unchanged.

Probably just need to do:

```
active_cells = active_from_xyz(mesh, topo, gridLoc, method)
return np.arange(mesh.n_cells)[active_cells]
```

File: SimPEG/utils/__init__.py [hunk 2]
```diff
@@ -104,6 +104,8 @@
 .. autosummary::
   :toctree: generated/
 
+  active_from_xyz
```
Comment (jcapriot) [inline SimPEG/utils/__init__.py:107]: Not sure what other's opinions are, but mine is that I don't necessarily like importing things defined in `discretize.utils` into `SimPEG.utils` just to shadow them there. I would prefer for the user to import them from `discretize` as needed.

Just trying to say avoid importing *more* things from `discretize` than we already are.

File: SimPEG/utils/model_utils.py [hunk 3]
```diff
@@ -316,6 +139,14 @@ def depth_weighting(mesh, reference_locs, indActive=None, exponent=2.0, threshol
     'exponent' and 'threshold' are two adjustable parameters.
     """
 
+    if "indActive" in kwargs:
+        warnings.warn(
+            "The indActive keyword argument has been deprecated, please use grid_loc. "
```
Comment (jcapriot) [inline SimPEG/utils/model_utils.py:144]: grid_loc -> active_cells

File: SimPEG/utils/mesh_utils.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import numpy as np
 
 from discretize.utils import (  # noqa: F401
+    random_model,
```
Comment (jcapriot) [inline SimPEG/utils/mesh_utils.py:4]: Any reason to add the random model import from `discretize` here?

File: SimPEG/utils/__init__.py [hunk 5]
```diff
@@ -104,6 +104,7 @@
 .. autosummary::
   :toctree: generated/
 
+  random_model
```
Comment (jcapriot) [inline SimPEG/utils/__init__.py:107]: I don't think this was imported anywhere in the __init__.py

File: SimPEG/utils/__init__.py (hunk 2 above)
Comment (dccowan) [reply SimPEG/utils/__init__.py]: I agree that it's weird to import the same things from two places. If it's a discretize function, it should be imported from discretize. And if it's SimPEG functionality, it should be imported from SimPEG.

File: SimPEG/utils/mesh_utils.py (hunk 4 above)
Comment (dccowan) [reply SimPEG/utils/mesh_utils.py:4]: I can't see why. I'm happy to remove it

Comment (dccowan) [reply SimPEG/utils/mesh_utils.py:4]: I would also consider moving it from discretize to SimPEG

Comment (dccowan) [reply SimPEG/utils/mesh_utils.py:4]: This can be done in a separate PR

**Comment by domfournier** [comment]
Well done gang. Thanks for this.

**Pull request merged**